packages/next/src/server/lib/router-utils/filesystem.ts

```diff
@@ -489,6 +500,22 @@ export async function setupFsCheck(opts: {
         }
       }
 
+      if (
+        opts.dev &&
+        // Strip metadata suffix (-\w{6}) to match the static metadata file convention.
+        isStaticMetadataFile(itemPath.replace(/-\w{6}(?=\.[^/]*$)/, ''))
```

Comment:
You're checking the static metadata file like pathnames, not the file itself, you should avoid using `isStaticMetadataFile` it's for actual filename not final pathname.
You can use existing helper `isMetadataRouteFile(.., [], false)`. Manually stripping a regex here kinda lost the benefits of using helper since you also implement part of it yourself.